test/framework/provider/wait.go

```diff
@@ -52,27 +61,35 @@ func WaitForAllProvidersInstalled(ctx context.Context, c client.Client, interval
 }
 
 // Wait for Provider to be successfully updated.
-func WaitForRevisionTransition(ctx context.Context, c client.Client, p2 string, p1 string, interval time.Duration, timeout time.Duration) error {
+func WaitForRevisionTransition(ctx context.Context, c client.Client, p2 string, p1 string, interval time.Duration, timeout time.Duration, t *testing.T) error {
 	if err := wait.PollImmediate(interval, timeout, func() (bool, error) {
 		l := &v1.ProviderRevisionList{}
 		if err := c.List(ctx, l); err != nil {
 			return false, err
 		}
+
 		// There should be a revision present for the initial revision and the upgrade.
 		if len(l.Items) != 2 {
+			t.Log("The no. of provider revisions is not equal to 2")
+			for i, item := range l.Items{
+				t.Logf("Provider revision %v : %v uses package %v", i, item.Name, item.Spec.Package)
+			}
 			return false, nil
 		}
-		for _, p := range l.Items {
+		for _, item := range l.Items {
 			// New ProviderRevision should be Active.
-			if p.Spec.Package == p2 && p.GetDesiredState() != v1.PackageRevisionActive {
+			if item.Spec.Package == p2 && item.GetDesiredState() != v1.PackageRevisionActive {
+				t.Logf("The state of new provider revision %v built from package %v and having version %v is %v", item.Name, item.Spec.Package, item.Spec.Revision, item.Status)
 				return false, nil
 			}
 			// Old ProviderRevision should be Inactive.
-			if p.Spec.Package == p1 && p.GetDesiredState() != v1.PackageRevisionInactive {
+			if item.Spec.Package == p1 && item.GetDesiredState() != v1.PackageRevisionInactive {
+				t.Logf("The state of old provider revision %v built from package %v and having version %v is %v", item.Name, item.Spec.Package, item.Spec.Revision, item.Status)
 				return false, nil
 			}
 			// Both ProviderRevisions should be healthy.
-			if p.GetCondition(v1.TypeHealthy).Status != corev1.ConditionTrue {
+			if item.GetCondition(v1.TypeHealthy).Status != corev1.ConditionTrue {
+				t.Logf("The condition of provider revision %v: %v built from package %v  is %v", item.Name, item.Spec.Revision, item.Spec.Package, item.Status.ConditionedStatus)
```

Comment:
can you make the format of this logging statement match the other 2? the arguments are reversed in this one, but keeping them consistent would be easier to read/parse in the logs if the test goes through more than 1 of these temporary failure/waiting conditions

```suggestion
				t.Logf("The condition of provider revision %v built from package %v and having version %v is %v", item.Name, item.Spec.Package, item.Spec.Revision, item.Status.ConditionedStatus)
```

Reply:
Done!